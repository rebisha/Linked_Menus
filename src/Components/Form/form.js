import React, { Component } from "react";
// component
import Button from "../Button/button";

class Form extends Component {
  constructor(props) {
    super(props);

    this.state = {
      userId: "",
      title: "",
      body: ""
    };
  }

  handleChange = e => {
    const { name, value } = e.target;

    this.setState({
      [name]: value
    });
  };

  submitForm = e => {
    e.preventDefault();

    const response = fetch("https://jsonplaceholder.typicode.com/posts", {
      method: "POST",
      mode: "cors",
      cache: "no-cache",
      credentials: "same-origin",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(this.state)
    }).catch(error => console.log("Error", error));

    this.refs.userId.value = null;
    this.refs.title.value = null;
    this.refs.body.value = null;
  };

  render() {
    const { userId, title, body } = this.state;

    return (
      <form className="form-wrapper" onSubmit={this.submitForm}>
        <label>User Id:</label>
        <input
          type="text"
          name="userId"
          ref="userId"
          value={userId}
          onChange={this.handleChange}
        />
        <label>Title:</label>
        <input
          type="text"
          name="title"
          ref="title"
          value={title}
          onChange={this.handleChange}
        />
        <label>Body:</label>
        <input
          type="text"
          name="body"
          ref="body"
          value={body}
          onChange={this.handleChange}
        />
        <div className="form-btn">
          <Button className="tertiary-menu" type="submit" name="Submit" />
          <Button
            className="tertiary-menu"
            type="reset"
            name="Cancel"
            onClick={this.props.handleCancel}
          />
        </div>
      </form>
    );
  }
}

export default Form;
