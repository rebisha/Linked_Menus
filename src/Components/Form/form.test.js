import React from "react";
import Form from "./form";
import renderer from "react-test-renderer";

test("Form component renders correctly", () => {
  const component = renderer.create(
    <Form handleSubmit={() => console.log("Submit form")} />
  );
  expect(component).toMatchSnapshot();
});
