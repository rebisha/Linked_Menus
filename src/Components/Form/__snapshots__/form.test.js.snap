// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders correctly 1`] = `
<form
  className="form-wrapper"
  onSubmit={[Function]}
>
  <label>
    User Id:
  </label>
  <input
    name="userId"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label>
    Title:
  </label>
  <input
    name="title"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label>
    Body:
  </label>
  <input
    name="body"
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="form-btn"
  >
    <button
      type="submit"
    >
      Submit 
    </button>
    <button
      type="reset"
    >
      Cancel
    </button>
  </div>
</form>
`;
